tests/units/test_remotes.rb

```diff
@@ -190,5 +190,25 @@ def test_push
     end
   end
 
+  def test_push_with_block
+    in_temp_dir do |path|
+      loc = Git.clone(@wbare, 'local')
+      rem = Git.clone(@wbare, 'remote', :config => 'receive.denyCurrentBranch=ignore')
 
+      loc.add_remote('testrem', rem)
+
+      loc.chdir do
+        new_file('test-file', 'blahblahblah')
+        loc.add
+        loc.commit('master commit')
+      end
+
+      block_called = false
+      loc.push('testrem') do |output|
+        assert_match(/master -> master/, output.read, 'expected block to receive output of push command.')
+        block_called = true
+      end
```

Comment:
I am not convinced, yet. Why not just let the `push` method return the output? Does it already return something different? (I admit, I haven't looked into the details.)

Reply:
As I said to @tarcinil, given that `command()` takes an optional block, why shouldn't all invokers of `command()` pass along an optional block?

Reply:
@robkinyon-roivant Sure, it would be possible. Would you care to write a PR for it, and also add some practical real-world examples of why and when it would be useful?

If it's _only_ for people writing tests, I am a bit more hesitant to make the change than if it's for general usability of the gem. I am not saying "no", I am just saying "maybe". All added code makes things more complex => each change have to be weighed towards "is it worth the  [potentially] decreased readability and [potentially] increased complexity", that's all.

Reply:
`Git::Lib#command` already returns the output of the command, so `push` would as well. The block here eventually gets passed down to `IO.popen` and provides access to the stdout stream of the git command as it runs. 
We use this library for pushing changes to a remote PaaS git server that runs some lifecycle hooks on an incoming push and it was useful to be able to view output as it happens rather than just getting a timeout (and potentially losing the output data).

Reply:
Fair enough, I'm willing to accept this @bsedat @robkinyon-roivant.

Please submit a PR that fixes it for all invokers of `command`, or was this already the case?